internal/services/projects.go

```diff
@@ -215,25 +221,36 @@ func updateProject(
 	}
 
 	if projectGroups != "" {
-		err = UpsertProjectGroups(groupsWrapper, &projModel, projectsWrapper, accessManagementWrapper, projModelResp, projectGroups, projectID, projectName, featureFlagsWrapper)
+		err = UpsertProjectGroupsByUpdateFlow(groupsWrapper, &projModel, projectsWrapper, accessManagementWrapper, projModelResp, projectGroups, projectID, projectName, featureFlagsWrapper)
 		if err != nil {
 			return projectID, err
 		}
 	}
 	return projectID, nil
 }
 
-func UpsertProjectGroups(groupsWrapper wrappers.GroupsWrapper, projModel *wrappers.Project, projectsWrapper wrappers.ProjectsWrapper,
+func UpsertProjectGroupsByCreateFlow(groupsWrapper wrappers.GroupsWrapper, projModel *wrappers.Project, projectsWrapper wrappers.ProjectsWrapper,
```

Comment:
related to next comment, what is the difference between the update and the create flow?

Reply:
in the update we should call first to GetGroupMap, and than continue calling to UpsertProjectGroups
but in create flow we did it at the creation part at the begin of the flow, so no need to call this logic twice

Reply:
But your first call in UpsertProjectGroups is to GetGroupMap func and both of the upsert flows (update/create) are calling UpsertProjectGroups function

Reply:
that what I wrote there - my bad forgot to remove this call
fixed